
error 1: FATAL FAULT `precondition`: debug: (numeric.this **! other)
Call stack:
fuzion.type.runtime.type.fault.type.install_default.λ.call#1: {base.fum}/fuzion/runtime/fault.fz:38:7:
      fuzion.sys.fatal_fault kind msg
------^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fuzion.runtime.fault.cause#1: {base.fum}/eff/fallible.fz:35:6:
  => h e
-----^
fuzion.type.runtime.type.contract_fault.type.install_default.λ.call#1: {base.fum}/fuzion/runtime/contract_fault.fz:39:7:
      fuzion.runtime.fault.cause kind_and_msg).default
------^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fuzion.runtime.contract_fault.cause#1: {base.fum}/eff/fallible.fz:35:6:
  => h e
-----^
fuzion.type.runtime.type.pre_fault.type.install_default.λ.call#1: {base.fum}/fuzion/runtime/pre_fault.fz:39:7:
      fuzion.runtime.contract_fault.cause ("precondition", msg)).default
------^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
fuzion.runtime.pre_fault.cause#1: {base.fum}/eff/fallible.fz:35:6:
  => h e
-----^
fuzion.runtime.precondition_fault#1: {base.fum}/fuzion/runtime/pre_fault.fz:58:47:
public precondition_fault(msg String) void => pre_fault.cause msg
----------------------------------------------^^^^^^^^^^^^^^^^^^^
i32.pre infix **: {base.fum}/numeric.fz:78:7:
      debug: (numeric.this **! other)
------^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
i32.precall infix **: {base.fum}/numeric.fz:77:5:
    pre
----^^^
      debug: (numeric.this **! other)
------^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
idiom32ex2.exp#3 i32: {main.fum}/-:3:41:
  exp     (T type : integer, x, n T) => x **  n
----------------------------------------^^^^^^^
idiom32ex2.loop.loop: {main.fum}/-:19:26:
      say "exp $i $j is {exp i j}"
-------------------------^^^^^^^
idiom32ex2.loop: {main.fum}/-:18:5:
    for j in 0..15 do
----^
idiom32ex2: {main.fum}/-:17:3:
  for i in 0..15 do
--^

*** fatal errors encountered, stopping.
one error.
